### ISSUE TITLE
[Pay meow][$500] [Held requests] Pay with Expensify button does not include amount after approving held request

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#39342]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
issue found when validating #33124
**Version Number:** 1.4.58-2
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:** n/a
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** applause internal team
**Slack conversation:**

## Action Performed:
Precondition:
- Admin and employee are in the Collect workspace.

1. [Employee] Request money from the workspace chat.
2. [Employee] Create a money request
3. [Employee] Hold the request.
4. [Admin] Go to expense report.
5. [Admin] Click Approve button.
6. [Admin] Approve all the request.
## Expected Result:
After approving all the held request, Pay with Expensify button will update to include the total amount (this is the case when approving two requests where one of the requests is held).
## Actual Result:
After approving all the held request, Pay with Expensify button does not include the total amount. It only includes the total amount after visiting the transaction thread.

## Workaround:
unknonw

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/7ed8f546-48b3-45e5-b925-f89d8a1955be

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e05ae27f1f02458c</li>
        <li>Upwork Job ID: 1774870331469250560</li>
        <li>Last Price Increase: 2024-04-01</li>
<li>Automatic offers: </li>
<ul>
<li>hoangzinh | Reviewer | 0</li>
</ul></li>
    </ul>
</details>

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
After approving all the held request, Pay with Expensify button does not include the total amount. It only includes the total amount after visiting the transaction thread.

### What is the root cause of that problem?
We set the `formattedAmount` prop as empty string if all transactions are held. After we approve the money request, the hold reason isn't cleared in transaction so this check is still `true` and `Pay with Expensify` button doesn't include amount.

When we open the transaction thread report, `OpenReport` is called and it returns the hold reason of transaction as empty. It leads this check is `false` and then the amount appears after going back from transaction thread report.

https://github.com/Expensify/App/blob/c848cb6bf29b600eecb8b338222fffe8a36d0329/src/components/MoneyReportHeader.tsx#L155
### What changes do you think we should make in order to solve the problem?
Since all hold transactions will be un-held after we approve the money request, we can add the check approve report here to display the amount if the report is approved
```
formattedAmount={(!ReportUtils.hasOnlyHeldExpenses(moneyRequestReport.reportID) || ReportUtils.isReportApproved(moneyRequestReport))? displayedAmount : ''}
```


https://github.com/Expensify/App/blob/c848cb6bf29b600eecb8b338222fffe8a36d0329/src/components/MoneyReportHeader.tsx#L155

https://github.com/Expensify/App/blob/c848cb6bf29b600eecb8b338222fffe8a36d0329/src/components/MoneyReportHeader.tsx#L187
### What alternative solutions did you explore? (Optional)
In `approveMoneyRequest` function, we can set the hold comment of all transactions to empty in `optimisticData` and then 

revert it to the current hold comment in `failureData`
https://github.com/Expensify/App/blob/c848cb6bf29b600eecb8b338222fffe8a36d0329/src/libs/actions/IOU.ts#L4609

**Comment by neil-marcellini** [comment]
@BartoszGrajdek are you around to help fix this?

**Comment by neil-marcellini** [comment]
@robertjchen would you be willing to take this over since you reviewed the PR and have the most context? I'll keep working on it until I hear otherwise.

**Comment by robertjchen** [comment]
sounds good, I can take it! This can definitely be handled by contributors 👍

**Comment by neil-marcellini** [comment]
> The expected result is that the pay button should not display the amount when all requests are on hold.
> The actual result is that it displays the amount after viewing the transaction thread.

Never mind, I see that this is wrong because fully approving a report removes the hold on all transactions. So the amount should be shown in the pay button.

**Comment by mallenexpensify** [comment]
I believe so. @hoangzinh , please review .

**Comment by robertjchen** [comment]
Looks good to me as well, let's move forward with @nkdengineer 's proposal 👍

**Comment by nkdengineer** [comment]
@hoangzinh The PR is [here](https://github.com/Expensify/App/pull/39579).

**Comment by nkdengineer** [comment]
@mallenexpensify The PR is deployed to production last month. Can you please add weekly label again and I think we're ready for payment here? Thanks.

**Comment by mallenexpensify** [comment]
Contributor: @nkdengineer owed $500 via Upwork
Contributor+: @hoangzinh paid $500 via Upwork. 

@nkdengineer can you please accept the job and reply here once you have? 

[PR hit production on 4/22,](https://github.com/Expensify/App/pull/39579#issuecomment-2069772245) automation failed here, I updated labels and such.

**Comment by nkdengineer** [comment]
@mallenexpensify Offer accepted, thanks 🙏

**Comment by mallenexpensify** [comment]
@nkdengineer paid, updated main payment comment above. 

@hoangzinh can you fill out the BZ checklist? Looks like it didn't auto-post here, pasted below 

**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@] Determine if we should create a regression test for this bug.
- [ ] [@] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@BZ] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:l

**Comment by mallenexpensify** [comment]
Thanks @hoangzinh , TestRail GH created
- https://github.com/Expensify/Expensify/issues/394735
